tests/checkpoint/test_v1_checkpoint.py

```diff
@@ -494,20 +494,23 @@ def test_checkpoint_run_no_actions(self, validation_definition: ValidationDefini
     def test_checkpoint_run_actions(
         self,
         validation_definition: ValidationDefinition,
-        actions: list[CheckpointAction],
+        mocker: MockerFixture,
     ):
-        validation_definitions = [validation_definition]
+        # Arrange
+        action = mocker.Mock(spec=UpdateDataDocsAction)
         checkpoint = Checkpoint(
             name=self.checkpoint_name,
-            validation_definitions=validation_definitions,
-            actions=actions,
+            validation_definitions=[validation_definition],
+            actions=[action],
         )
 
-        with mock.patch.object(ValidationAction, "v1_run") as mock_run:
-            result = checkpoint.run()
+        # Act
+        result = checkpoint.run()
 
-        assert mock_run.call_count == len(actions)
-        mock_run.assert_called_with(checkpoint_result=result, action_context=mock.ANY)
+        # Assert
+        action._copy_and_set_values().v1_run.assert_called_once_with(
```

Comment:
`action._copy_and_set_values()` <- not loving reaching into private mock methods, but probably not the end of the world. This is something having to do with pydantic cloning the mock?

Reply:
Ugh yeah I'm researching alternative ways to deal with this but I'm not sure